From 6210f3dca2ff0b8bdb581d1382887c4f6a8acc7d Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Mon, 17 Jun 2013 02:29:07 +0200
Subject: [PATCH 609/715] drm/rcar-du: Use devm_ioremap_resource()

Replace the devm_request_mem_region() and devm_ioremap_nocache() calls
with devm_ioremap_resource().

Signed-off-by: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
(cherry picked from commit 4e898cdb89d4fbea55a3aee9d76889164d03b167)

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit ec0b0454f2cb96a9ed06e6e705412d3ba5f9db4e)

Conflicts:
	drivers/gpu/drm/rcar-du/rcar_du_drv.c

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_drv.c |   11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_drv.c b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
index 982f7d4..4f54a7a 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_drv.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
@@ -135,7 +135,6 @@ static int rcar_du_load(struct drm_device *dev, unsigned long flags)
 	struct platform_device *pdev = dev->platformdev;
 	struct rcar_du_platform_data *pdata = pdev->dev.platform_data;
 	struct rcar_du_device *rcdu;
-	struct resource *ioarea;
 	struct resource *lvds0_ioarea;
 	struct resource *lvds1_ioarea;
 	struct resource *mem;
@@ -166,15 +165,7 @@ static int rcar_du_load(struct drm_device *dev, unsigned long flags)
 		return -EINVAL;
 	}
 
-	ioarea = devm_request_mem_region(&pdev->dev, mem->start,
-					 resource_size(mem), pdev->name);
-	if (ioarea == NULL) {
-		dev_err(&pdev->dev, "failed to request memory region\n");
-		return -EBUSY;
-	}
-
-	rcdu->mmio = devm_ioremap_nocache(&pdev->dev, ioarea->start,
-					  resource_size(ioarea));
+	rcdu->mmio = devm_ioremap_resource(&pdev->dev, mem);
 	if (rcdu->mmio == NULL) {
 		dev_err(&pdev->dev, "failed to remap memory resource\n");
 		return -ENOMEM;
-- 
1.7.10.4

